src/rust/generators/sysmon-generator/tests/integration_test.rs

```diff
@@ -0,0 +1,228 @@
+        tracing_subscriber::registry()
+            .with(filter)
+            .with(log_layer)
+            .with(tracing_opentelemetry::layer().with_tracer(tracer))
+            .init();
+
+        tracing::info!("logger configured successfully");
+
+        let endpoint = std::env::var("PIPELINE_INGRESS_CLIENT_ADDRESS")
+            .expect("missing environment variable PIPELINE_INGRESS_CLIENT_ADDRESS");
+
+        let bootstrap_servers = std::env::var("KAFKA_BOOTSTRAP_SERVERS")
+            .expect("missing environment variable KAFKA_BOOTSTRAP_SERVERS");
+        let sasl_username = std::env::var("INTEGRATION_TESTS_KAFKA_SASL_USERNAME")
+            .expect("missing environment variable INTEGRATION_TESTS_KAFKA_SASL_USERNAME");
+        let sasl_password = std::env::var("INTEGRATION_TESTS_KAFKA_SASL_PASSWORD")
+            .expect("missing environment variable INTEGRATION_TESTS_KAFKA_SASL_PASSWORD");
+        let consumer_group_name = std::env::var("INTEGRATION_TESTS_KAFKA_CONSUMER_GROUP_NAME")
+            .expect("missing environment variable INTEGRATION_TESTS_KAFKA_CONSUMER_GROUP_NAME");
+
+        tracing::info!(
+            message = "waiting 10s for pipeline-ingress to report healthy",
+            endpoint = %endpoint,
+        );
+
+        HealthcheckClient::wait_until_healthy(
+            endpoint.clone(),
+            "graplinc.grapl.api.pipeline_ingress.v1beta1.PipelineIngressService",
+            Duration::from_millis(10000),
+            Duration::from_millis(500),
+        )
+        .await
+        .expect("pipeline-ingress never reported healthy");
+
+        tracing::info!("connecting pipeline-ingress gRPC client");
+        let pipeline_ingress_client = PipelineIngressClient::connect(endpoint.clone())
+            .await
+            .expect("could not configure gRPC client");
+
+        SysmonGeneratorTestContext {
+            pipeline_ingress_client,
+            bootstrap_servers,
+            sasl_username,
+            sasl_password,
+            consumer_group_name,
+            _guard,
+        }
+    }
+}
+
+#[test_context(SysmonGeneratorTestContext)]
+#[tokio::test]
+async fn test_sysmon_event_produces_expected_graph(ctx: &mut SysmonGeneratorTestContext) {
+    let event_source_id = Uuid::new_v4();
+    let tenant_id = Uuid::new_v4();
+
+    let bootstrap_servers = ctx.bootstrap_servers.clone();
+    let sasl_username = ctx.sasl_username.clone();
+    let sasl_password = ctx.sasl_password.clone();
+    let consumer_group_name = ctx.consumer_group_name.clone();
+
+    tracing::info!("configuring kafka consumer");
+    let kafka_consumer = Consumer::new(
+        bootstrap_servers,
+        sasl_username,
+        sasl_password,
+        consumer_group_name,
+        "generated-graphs".to_string(),
+    )
+    .expect("could not configure kafka consumer");
+
+    tracing::info!("creating kafka subscriber thread");
+    let kafka_subscriber = tokio::task::spawn(async move {
```

Comment:
You probably want to add a `.instrument()` to the async block for tracing

Reply:
done